plantcv/plantcv/roi/roi_methods.py

```diff
@@ -194,7 +193,156 @@ def _draw_roi(img, roi_contour):
+    overlap_img = np.zeros((height, width))
+    # Initialize a binary image of the circle that will contain all ROI
+    all_roi_img = np.zeros((height, width), dtype=np.uint8)
+    roi_objects = Objects()
+    for i in range(0, len(coord)):
+        # Initialize a binary image for each circle
+        bin_img = np.zeros((height, width), dtype=np.uint8)
+        y = coord[i][1]
+        x = coord[i][0]
+        # Draw the circle on the binary image
+        # Keep track of all roi
+        all_roi_img = cv2.circle(all_roi_img, (x, y), radius, 255, -1)
+        # Keep track of each roi individually to check overlapping
+        circle_img = cv2.circle(bin_img, (x, y), radius, 255, -1)
+        overlap_img = overlap_img + circle_img
+        # Make a list of contours and hierarchies
+        rc, rh = cv2.findContours(circle_img, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE)[-2:]
+        roi_objects.append(rc, rh)
+    return roi_objects, overlap_img, all_roi_img
+
+
+def _grid_roi(img, nrows, ncols, coord=None, radius=None, spacing=None):
+    if radius is None:
+        RADIUS_RATIO = 0.325
+        if spacing[0] == 0:
+            radius = round(RADIUS_RATIO*spacing[1])
+        elif spacing[1] == 0:
+            radius = round(RADIUS_RATIO*spacing[0])
+        else:
+            radius = round(RADIUS_RATIO*(spacing[0]+spacing[1])/2)
+    # Get the height and width of the reference image
+    height, width = np.shape(img)[:2]
+    radius = _adjust_radius_grid(height, width, coord, radius, spacing, nrows, ncols)
+    overlap_img = np.zeros((height, width))
+    # Initialize a binary image of the circle that will contain all ROI
+    all_roi_img = np.zeros((height, width), dtype=np.uint8)
+    roi_objects = Objects()
+    # Loop over each row
+    for i in range(0, nrows):
+        # The upper left corner is the y starting coordinate + the ROI offset * the vertical spacing
+        y = coord[1] + i * spacing[1]
+        # Loop over each column
+        for j in range(0, ncols):
+            # Initialize a binary image for each circle
+            bin_img = np.zeros((height, width), dtype=np.uint8)
+            # The upper left corner is the x starting coordinate + the ROI offset * the
+            # horizontal spacing between chips
+            x = coord[0] + j * spacing[0]
+            # Draw the circle on the binary images
+            # Keep track of all roi
+            all_roi_img = cv2.circle(all_roi_img, (x, y), radius, 255, -1)
+            # Keep track of each roi individually to check overlapping
+            circle_img = cv2.circle(bin_img, (x, y), radius, 255, -1)
+            overlap_img = overlap_img + circle_img
+            # Make a list of contours and hierarchies
+            rc, rh = cv2.findContours(circle_img, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE)[-2:]
+            roi_objects.append(rc, rh)
+    return roi_objects, overlap_img, all_roi_img
+
+
+def auto_grid(bin_mask, nrows, ncols, radius=None, img=None):
```

Comment:
I suggest we refactor the input keyword `bin_mask`. We use two standardized keywords elsewhere, `bin_img` or `mask`. Admittedly they pretty much mean the same thing (maybe there's sort of a connotation that `mask` is a clean binary image vs `bin_img` is not necessarily filtered yet)